src/App.tsx

```diff
@@ -51,6 +52,25 @@ const App = () => {
       .getUiSchema()
       .then((uiSchema) => dispatch(Actions.setUiSchema(uiSchema)));
   }, [schemaService]);
+  useEffect(() => {
+    setSelection((oldSelection) => {
+      if (!oldSelection) {
+        return undefined;
+      }
+      const newSelectedUISchemaElement = uiSchema
+        ? findByUUID(uiSchema, oldSelection.uuid)
+        : undefined;
+
+      if (
+        isUUIDError(newSelectedUISchemaElement) ||
+        !newSelectedUISchemaElement
+      ) {
+        //element does not exist anymore - clear old selection
+        return undefined;
+      }
+      return oldSelection;
+    });
+  }, [uiSchema, schema]);
```

Comment:
`schema` no longer needs to be part of the dependencies
```suggestion
  }, [uiSchema]);
```